python/tests/src/test_api_object.py

```diff
@@ -0,0 +1,132 @@
+import unittest
+import mock
+from os.path import dirname, abspath, join
+import sys
+
+# required for imports in api_object to work
+sys.path.append(abspath(join(dirname(__file__), '..', 'src')))
+
+from api_common import SpamException
+from api_common import RateLimitException
+from api_object import ApiObject
+
+class ApiObjectTest(unittest.TestCase):
+
+    @mock.patch('src.api_object.requests.delete')
+    @mock.patch('src.api_object.requests.post')
+    @mock.patch('src.api_object.requests.put')
+    @mock.patch('src.api_object.requests.get')
```

Comment:
[`requests-mock`](https://github.com/jamielennox/requests-mock) is a nice(r) way to produce fake `requests` responses for tests.

Reply:
I backlogged a task in Jira to look at using requests-mock: https://jira.pinadmin.com/browse/TPP-836